### PR TITLE
Feat: add comments for config

### PR DESCRIPTION
**Pull request by watermelo**
**What this PR does**:
Add comments for config

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by watermelo** [comment]
#566

File: config/consumer_config.go [hunk 1]
```diff
@@ -75,17 +75,17 @@ func (c *ConsumerConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
...
 }
 
-// SetConsumerConfig ...
+// SetConsumerConfig sets consumerConfig by @c
 func SetConsumerConfig(c ConsumerConfig) {
 	consumerConfig = &c
 }
 
-// ConsumerInit ...
+// ConsumerInit Load config file to init consumer config
```
Comment (zouyx) [inline config/consumer_config.go:88]:
```suggestion
// ConsumerInit loads config file to init consumer config
```

File: config/graceful_shutdown.go [hunk 2]
```diff
@@ -126,10 +126,8 @@ func destroyConsumerProtocols(consumerProtocols *gxset.HashSet) {
 	}
 }
 
-/**
- * destroy the provider's protocol.
- * if the protocol is consumer's protocol too, we will keep it.
- */
+// destroy the provider's protocol.
```
Comment (zouyx) [inline config/graceful_shutdown.go:129]:
```suggestion
// destroyProviderProtocols destroys the provider's protocol.
```

File: config/interfaces/config_reader.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ package interfaces
 
 import "bytes"
 
-// ConfigReader
+// A ConfigReader interface is used to read config from consumer or provider
```
Comment (zouyx) [inline config/interfaces/config_reader.go:22]:
```suggestion
// ConfigReader is used to read config from consumer or provider
```

File: config/provider_config.go [hunk 4]
```diff
@@ -67,17 +67,17 @@ func (c *ProviderConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
...
 }
 
-// SetProviderConfig ...
+// SetProviderConfig sets provider config by @p
 func SetProviderConfig(p ProviderConfig) {
 	providerConfig = &p
 }
 
-// ProviderInit ...
+// ProviderInit load config file to init provider config
```
Comment (zouyx) [inline config/provider_config.go:80]:
```suggestion
// ProviderInit loads config file to init provider config
```

File: config/provider_config.go [hunk 5]
```diff
@@ -55,7 +55,7 @@ type ProviderConfig struct {
 	ConfigType        map[string]string          `yaml:"config_type" json:"config_type,omitempty" property:"config_type"`
 }
 
-// UnmarshalYAML ...
+// UnmarshalYAML unmarshal the ProviderConfig by @unmarshal function
```
Comment (AlexStocks) [inline config/provider_config.go:58]: unmarshal -> unmarshals

File: config/service_config.go [hunk 6]
```diff
@@ -80,12 +80,12 @@ type ServiceConfig struct {
...
 }
 
-// Prefix ...
+// Prefix returns ServiceConfig prefix
 func (c *ServiceConfig) Prefix() string {
 	return constant.ServiceConfigPrefix + c.InterfaceName + "."
 }
 
-// UnmarshalYAML ...
+// UnmarshalYAML unmarshal the ServiceConfig by @unmarshal function
```
Comment (AlexStocks) [inline config/service_config.go:88]: unmarshal -> unmarshals

File: config/config_center_config.go [hunk 7]
```diff
@@ -31,7 +31,13 @@ import (
 	"github.com/apache/dubbo-go/common/constant"
 )
 
-// ConfigCenterConfig ...
+// ConfigCenterConfig is configuration for config center
+//
+// ConfigCenter also introduced concepts of namespace and group to better manage Key-Value pairs by group,
+// those concepts are already built-in in many professional third-party configuration centers.
+// In most cases, namespace is used to isolate different tetants, while group is used to divid the key set from one tetant into groups.
```
Comment (Patrick0308) [inline config/config_center_config.go:38]:
```suggestion
// Those configs are already built-in in many professional third-party configuration centers.
// In most cases, namespace is used to isolate different tenants, while group is used to divide the key set from one tenant into groups.
```

**Comment by AlexStocks** [comment]
LGTM

File: config/consumer_config.go [hunk 8]
```diff
@@ -63,7 +63,7 @@ type ConsumerConfig struct {
 	ConfigType     map[string]string           `yaml:"config_type" json:"config_type,omitempty" property:"config_type"`
 }
 
-// UnmarshalYAML ...
+// UnmarshalYAML unmarshal the ConsumerConfig by @unmarshal function
```
Comment (zouyx) [inline config/consumer_config.go:66]:
```suggestion
// UnmarshalYAML unmarshals the ConsumerConfig by @unmarshal function
```

File: config/reference_config.go [hunk 9]
```diff
@@ -76,7 +76,7 @@ func NewReferenceConfig(id string, ctx context.Context) *ReferenceConfig {
 	return &ReferenceConfig{id: id, context: ctx}
 }
 
-// UnmarshalYAML ...
+// UnmarshalYAML unmarshal the ReferenceConfig by @unmarshal function
```
Comment (zouyx) [inline config/reference_config.go:79]:
```suggestion
// UnmarshalYAML unmarshals the ReferenceConfig by @unmarshal function
```

File: config/registry_config.go [hunk 10]
```diff
@@ -47,7 +47,7 @@ type RegistryConfig struct {
 	Params     map[string]string `yaml:"params" json:"params,omitempty" property:"params"`
 }
 
-// UnmarshalYAML ...
+// UnmarshalYAML unmarshal the RegistryConfig by @unmarshal function
```
Comment (zouyx) [inline config/registry_config.go:50]:
```suggestion
// UnmarshalYAML unmarshals the RegistryConfig by @unmarshal function
```

**Pull request merged**